### PR TITLE
Fix UART flush on the ESP32.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/uart_esp32.cc [hunk 1]
```diff
@@ -178,6 +178,7 @@ class UartResource : public EventQueueResource {
   UART_ISR_INLINE void enable_tx_interrupts(bool begin);
   UART_ISR_INLINE void disable_tx_interrupts(bool done);
 
+  UART_ISR_INLINE bool has_flushed_tx_fifo();
```
Comment (kasperl) [inline src/resources/uart_esp32.cc:181]: Perhaps `is_empty_tx_fifo()`? Somehow I think the "has flushed" could mean that it only returns true if you actively asked for flushing.

File: src/resources/uart_esp32.cc [hunk 2]
```diff
@@ -461,10 +462,14 @@ UART_ISR_INLINE void UartResource::disable_tx_interrupts(bool done) {
   }
 }
 
+UART_ISR_INLINE bool UartResource::has_flushed_tx_fifo() {
+  return get_tx_fifo_free() == SOC_UART_FIFO_LEN && uart_toit_hal_is_tx_idle(hal_);
```
Comment (kasperl) [inline src/resources/uart_esp32.cc:466]: Maybe add a comment that explains why we need to check for idle? It wasn't clear to the original writer of the code :)

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/resources/uart_esp32.cc (hunk 1 above)
Comment (floitsch) [reply src/resources/uart_esp32.cc]: done.

File: src/resources/uart_esp32.cc (hunk 2 above)
Comment (floitsch) [reply src/resources/uart_esp32.cc:470]: done.

**Pull request merged**